pilot/pkg/security/model/authentication.go

```diff
@@ -217,10 +227,17 @@ func ApplyToCommonTLSContext(tlsContext *tls.CommonTlsContext, metadata *model.N
 			CaCertificatePath: metadata.TLSServerRootCert,
 		}
 
+		// TODO: if subjectAltName ends with *, create a prefix match as well.
+		// TODO: if user explicitly specifies SANs - should we alter his explicit config by adding all spifee aliases?
+		matchSAN := util.StringToExactMatch(subjectAltNames)
+		if len(trustDomainAliases) > 0 {
+			matchSAN = append(matchSAN, util.StringToPrefixMatch(appendURIPrefixToTrustDomain(trustDomainAliases))...)
```

Comment:
This does not make sense to me. If I configure SAN=foo.com, previous we match foo.com only

Now we configure:
```yaml
exact: foo.com
prefix: spiffe://alias/
```

This seems broken...

Reply:
I guess this is by design only which was approved by both networking/security wg. I'm not sure what's there to fix here

Reply:
for sidecar the SAN is always empty (also I don't think we even support it on sidecar?), we don't support configuring it. for gateway, the trustDomainAliases is always empty as this PR only adds it for sidecar.

later when supporting gateway, if we are okay to let the user to manually specify the allowed trust domains, this could skip appending the trust domain prefix if SAN is non empty.

Reply:
This code is used for gateway as well as far as I can tell. Where is it selecting just sidecar?

Reply:
https://github.com/istio/istio/pull/26224/files/03539ee5b777d7bddcaab4fca5e2dd9af422ac90#diff-9eec7bb8f46726e51ab373e7b8e11ca7R452

Unless I'm missing something from istiod side, for Gateway's we always send an empty list

Reply:
in gateway.go, it always pass []string{} for trustDomainAliases.

Reply:
I see. This is extremely subtle, extremely security conscious, and untested... at the very least we need some clear comments here I guess